### PR TITLE
Add C90 support for helpers.h

### PR DESCRIPTION
**Pull request by axic**
Fixes #273.

### CODE REVIEW

File: include/evmc/helpers.h [hunk 1]
```diff
@@ -19,26 +19,39 @@
 
 #include <evmc/evmc.h>
 
+#define EVMC_INLINE
+#ifdef __cplusplus
+#  undef EVMC_INLINE
+#  define EVMC_INLINE inline
+#endif
+#if __STDC__
+#  if __STDC_VERSION__ >= 199901L
```
Comment (axic) [inline include/evmc/helpers.h]: Could also merge these into `if defined(__cplusplus) || __STD_VERSION__ >= 199901L` if we assume the preprocessor for C90-only compiler is clever enough...

Comment (chfast) [reply include/evmc/helpers.h]: This would work, because if `__STDC_VERSION__` is undefined it will be substituted with `0`.

Comment (axic) [reply include/evmc/helpers.h]: By the clever enough I meant that it supports the `defined()` expression as well as the `or` operator`.

**Comment by axic** [comment]
@chfast simplified detection, will squash the last commit into the first if we are going to merge this.

**Comment by chfast** [comment]
This is not useful at all and your are wasting your time.

**Comment by axic** [comment]
> This is not useful at all and your are wasting your time.

Thanks! I don't get the point why you are saying EVMC in C90 compatible when it isn't :D

I've suggested in the previous PR that we state it is not C90 compatible.

**Comment by chfast** [comment]
> I've suggested in the previous PR that we state it is not C90 compatible.

Sorry, I missed that.
The `evmc.h` is. The rest is not as not being the core part of it. But we can remove the `C90` claim. The 20+ year old C standard does not matter.

**Comment by axic** [comment]
> The evmc.h is. The rest is not as not being the core part of it. But we can remove the C90 claim. The 20+ year old C standard does not matter.

I'd propose the README/docs just says "limited support for C90", the core header (evmc.h) is kept compatible, but that is not guaranteed for anything else.

**Pull request closed without merging**